### ISSUE TITLE
New value type (null) is not in annotated types. ( CallableParameterUseCaseInTypeContextInspection )

### ISSUE THREAD

**Issue by spenserhale** [kalessil/phpinspectionsea#836]
**Example Code 1**
```php
<?php
public static function appendInputClass( string $input, GF_Field $field ): string
{
    if (\in_array($field->get_input_type(), ['text', 'email', 'password', 'textarea', 'select'], true)) {
        $replacement = preg_replace("|class='(.*?)'|", "class='$1 form-control'", $input);

        if (null === $replacement) {
            $error = array_flip(get_defined_constants(true)['pcre'])[preg_last_error()];
            Rollbar::warning(__CLASS__ . '::' . __METHOD__ . ' preg replace error: ' . $error);
        } else {
            $input = $replacement;
        }
    }

    return $input;
}
```
**Example Code 2**
```php
<?php
public static function appendInputClass( string $input, GF_Field $field ): string
    {
        if (\in_array($field->get_input_type(), ['text', 'email', 'password', 'textarea', 'select'], true)) {
            $replacement = preg_replace("|class='(.*?)'|", "class='$1 form-control'", $input);

            if (null !== $replacement) {
                $input = $replacement;
            } else {
                $error = array_flip(get_defined_constants(true)['pcre'])[preg_last_error()];
                Rollbar::warning(__CLASS__ . '::' . __METHOD__ . ' preg replace error: ' . $error);
            }
        }

        return $input;
    }
```
Examples 1 & 2
The problem line is $input = $replacement.
I am specifically checking for null and handle error, but check still thinks value is null.

**Example 3**
```php
<?php
public static function appendInputClass( string $input, GF_Field $field ): string
{
    if (\in_array($field->get_input_type(), ['text', 'email', 'password', 'textarea', 'select'], true)) {
        $replacement = preg_replace("|class='(.*?)'|", "class='$1 form-control'", $input);

        if (\is_string($replacement)) {
                $input = $replacement;
            } else {
                $error = array_flip(get_defined_constants(true)['pcre'])[preg_last_error()];
                Rollbar::warning(__CLASS__ . '::' . __METHOD__ . ' preg replace error: ' . $error);
            }
    }

    return $input;
}
```

Example 3 works, and correctly does not throw inspection warning, however this method is ran from a loop and would like to reduce function calls when not needed.

**Comment by kalessil** [comment]
Thank you for reporting @spenserhale, I'll check what can we do here (to be specific why null is not filtered out - can be both IDE or my plugin).

**Comment by kalessil** [comment]
Fixed
